packages/golden_toolkit/lib/src/multi_screen_golden.dart

```diff
@@ -25,6 +30,9 @@ Future<void> _onlyPumpAndSettle(WidgetTester tester) async =>
 ///
 /// [customPump] optional pump function, see [CustomPump] documentation
 ///
+/// [deviceSetupBetweenSizeChanges] allows custom setup after the window changes size and before the images are rendered.
```

Comment:
```suggestion
/// [deviceSetupBetweenSizeChanges] allows custom setup when the window size is being changed.
```

Reply:
With this suggested change I would have no idea why I would need to modify this or customPump.

Reply:
I'm not following what images we are talking about..

Also would be great to have a test / example for custom setup

Reply:
Also I'm confused now since before we've been using / explaining our CustomPump as a tool for custom pumps, but now the message is different..
Should we rename CustomPump into CustomTesterSetup?

Reply:
I'm still struggling with the name here. I really don't like having the second parameter, but I discussed with Aaron and couldn't come up with an alternative.

Ultimately, I think the right near term answer is to actually deprecate screenMatchesGolden AND multiScreenGolden

and replace with an extension on WidgetTester:
```tester.screenMatchesGolden()``` that combines the functionality of both (essentially, augment the existing ```screenMatchesGolden()``` method to take an optional devices parameter (and default it to a single device). I suspect there is a way to collapse these two arguments and adjust the sequencing of when they get called.

Let's chat about it tomorrow.

Reply:
I also want to second Max. I think we should have a use case captured in a test that justifies this. Not only will it help document the intent, but it will serve as regression so we can safely refactor.

@aanelson do you think you could devise a simple example (requiring a couple ticks of the run loop before requesting images after a resize?)

Reply:
could we rename this to simply "deviceSetup"?